src/secureSignalUid2.ts

```diff
@@ -17,9 +17,17 @@ export function __uid2SSProviderScriptLoad() {
     callbacks: [],
   };
   window.__uid2.callbacks?.push((eventType) => {
-    //@ts-ignore
-    if (eventType === 'SdkLoaded') {
-      window.__uid2SecureSignalProvider!.registerSecureSignalProvider();
+    if (
+      eventType === 'InitCompleted' ||
+      eventType === 'SdkLoaded' ||
+      eventType === 'IdentityUpdated'
+    ) {
+      if ('getIdentity' in window.__uid2! && window.__uid2!.getIdentity()) {
+        if (eventType === 'IdentityUpdated') {
+          window.__uid2SecureSignalProvider?.resetProviderRegistration();
```

Comment:
This will reset the flag, but that means we'll register the collector again whenever identity is updated - is that OK? You mentioned registering it multiple times can cause problems with the bidder.